src/riak_kv_qry.erl

```diff
@@ -118,8 +257,10 @@ maybe_await_query_results(_) ->
     % we can't use a gen_server call here because the reply needs to be
     % from an fsm but one is not assigned if the query is queued.
     receive
-        Result ->
-            Result
+        {ok, Result} ->
```

Comment:
You are simply guarding against bad results? For proper responses, this should be the same.